File: aoc-2022/day-03/python/part2_sol.py
f = open("../input.txt")
data = f.read()

rucksacks = data.split("\n")

sum = 0
for i in range(0, len(rucksacks), 3):
    for item in rucksacks[i]:
        if (item in rucksacks[i+1] and item in rucksacks[i+2]):
            if ord(item) >= 97:
                sum += ord(item) - 96
            else:
                sum += ord(item) - 64 + 26
            break

print(sum)

File: aoc-2022/day-03/rust/part2_sol.rs
use std::fs;

fn main() {
    let data = fs::read_to_string("../input.txt").expect("Couldn't read from file");
    let mut sum = 0;
    let rucksacks: Vec<&str> = data.lines().collect();
    let l = rucksacks.len();
    for i in (0..l).step_by(3) {
        for item in rucksacks[i].chars() {
            let v = rucksacks[i+1].chars().collect::<Vec<char>>();
            let w = rucksacks[i+2].chars().collect::<Vec<char>>();
            if v.contains(&item) && w.contains(&item) {
                let ord = item as u32;
                if ord > 97 {
                    sum += ord - 96;
                } else {
                    sum += ord - 64 + 26;
                }
                break;
            }
        }
    }

    println!("{}", sum);
}